### ISSUE TITLE
Support indexConfigs in google_logging_project_bucket_config

### ISSUE THREAD

**Issue by i1snow** [hashicorp/terraform-provider-google#13287]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment. If the issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If the issue is assigned to a user, that user is claiming responsibility for the issue. If the issue is assigned to "hashibot", a community member has claimed the issue already.

### Description

According to https://cloud.google.com/logging/docs/reference/v2/rest/v2/billingAccounts.locations.buckets#LogBucket, GCP allows to index a field in a log bucket for a fast query. However, there is no argument for the feature in `google_logging_project_bucket_config`. Our workaround has been updating the bucket through gcp cli.

### New or Affected Resource(s)

* google_logging_project_bucket_config

### Potential Terraform Configuration

This is from https://github.com/hashicorp/terraform-provider-google/issues/11572 and `indexConfigs` argument looks good to me
```tf
resource "google_logging_log_bucket" "example_bucket"{
  description   =  "Example logging bucket"
  retentionDays =  30
  locked        =  false
  cmekSettings {
    kmsKeyName        =  "my-kms-key"
    kmsKeyVersionName =  "1"
  }
  indexConfigs = [
    {
      fieldPath =  "someField.intValue"
      type      =  "INDEX_TYPE_INTEGER"
    },
    {
      fieldPath =  "someOtherField.stringValue"
      type      =  "INDEX_TYPE_STRING"
    }
  ]
  restrictedFields = [
    "someField",
    "someOtherField"
  ]
}
```

### References

* https://cloud.google.com/logging/docs/analyze/custom-index#api_1
* https://cloud.google.com/logging/docs/reference/v2/rest/v2/projects.locations.buckets

b/302672916

**Comment by mudim** [comment]
Hi @i1snow, this has been released as of TF provider version 5.6.0. There is now support for index configurations in [bucket config](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/logging_project_bucket_config#index_configs).
Thanks

**Comment by zach-broadway-nauto** [comment]
The documentation for this feature is incorrect. 
https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/logging_project_bucket_config#index_configs

The example code shows:
```
resource "google_logging_project_bucket_config" "example-project-bucket-index-configs" {
  project          = "project_id"
  location         = "global"
  retention_days   = 30
  bucket_id        = "custom-bucket"

  index_configs   = {
    file_path   = "jsonPayload.request.status"
    type        = "INDEX_TYPE_STRING"
  }
}
``` 
This example code should however be:
```
resource "google_logging_project_bucket_config" "example-project-bucket-index-configs" {
  project          = "project_id"
  location         = "global"
  retention_days   = 30
  bucket_id        = "custom-bucket"

  index_configs   {
    field_path   = "jsonPayload.request.status"
    type        = "INDEX_TYPE_STRING"
  }
}
``` 
Note changes on the following:
index_configs   = { 
to  
index_configs  {

and 
file_path
to
field_path
